### DISCUSSION TITLE
Security Option: Disallow any connections to client until RustDesk Client window is opened

### DISCUSSION THREAD

**Discussion by nevaran** [rustdesk/rustdesk#7033] [Q&A] [answered]
I would like to propose a (possibly) easy to implement solution to a few issues I have noticed.
Unless the program is installed, the UAC popups cannot really be controlled, which in some situations is needed; but the installation of RustDesk adds a 24/7 running service that, in other situations, is a bad thing to have.

I would like to propose a checkmark option in the security section to "Disallow connections unless client is open" option. That way the service can be running in the background but no one can connect to the device until the person requires some IT support for example, in which case they can just "Open" the RustDesk program in order to allow connections.

This is a workaround due to disabling/enabling the service itself would require admin priviledges - something people would not normally have in these situations. Meanwhile in devices that are only used for remote management can still have the option turned off in order to connect to them 24/7

**Comment by rustdesk** [comment]
This is a good example if you want a feature request get high priority.
https://github.com/rustdesk/rustdesk/discussions/7437

**Reply by Sovince37** [reply]
I made a subject witch has been deleted because of wrong format... I made a 2nd subject with the correct format witch has been deleted too because of duplicate...
But in the opened subject, there is no response...

So... what is the correct way to be read ?

**Reply by rustdesk** [reply]
You do not need to create any new one, I have understood you. I also know AnyDesk has this feature. This dicussion has covered your idea.

**Reply by Sovince37** [reply]
Ok thank you for your reply, and for your work.
any chance for the feature to be developed this year ?

**Reply by rustdesk** [reply]
This is a good example if you want a feature request get high priority.
https://github.com/rustdesk/rustdesk/discussions/7437

**Reply by rustdesk** [reply]
You know how long of the feature request list, https://github.com/rustdesk/rustdesk/discussions/categories/feature-request
Everyone take their feature request as most urgent.

**Comment by rustdesk** [comment, marked as answer]
https://twitter.com/rustdesk/status/1777703501671575941

**Reply by Sovince37** [reply]
YEAH !!!! THANK YOU !!!!!

**Comment by thenktor** [comment]
Hi,

actually I think the solution is not optimal. It's still better to just run the service, when it's needed. Examples why running the service 24/7 is a bad idea:

- A security issue in the service itself instantly affects all systems, that have RustDesk installed
- A hacked RustDesk server may grant access to all connected systems to the attacker

Therefore I'd really prefer to run the service only when remote support is needed.

The access permissions of the Windows service can easily be changed. The default permissions are:
```txt
PS> sc.exe sdshow RustDesk

D:(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)S:(AU;FA;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;WD)
```

To allow all authenticated users (AU) to start/stop, query, and control the service you have to change permissions like this:
```txt
PS> sc.exe sdset RustDesk 'D:(A;;CCLCSWRPWPDTLOCRRC;;;SY)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCLCSWLOCRRC;;;IU)(A;;CCLCSWLOCRRC;;;SU)(A;;CCLCSWRPWPLOCRRC;;;AU)S:(AU;FA;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;WD)'
[SC] SetServiceObjectSecurity SUCCESS
```

These SDDL strings are horrible to read, I suggest an AI service like CHatGPT to get them explained in seconds.

Permissions can be set during installation.
